schematic/store/synapse.py

```diff
@@ -1517,6 +1517,60 @@ async def store_async_annotation(self, annotation_dict: dict) -> Annotations:
         )
         return await annotation_class.store_async(synapse_client=self.syn)
 
+    def process_row_annotations(
+        self,
+        dmge: DataModelGraphExplorer,
+        metadata_syn: Dict[str, Any],
+        hide_blanks: bool,
+        csv_list_regex: str,
+        annos: Dict[str, Any],
+        annotation_keys: str,
+    ) -> Dict[str, Any]:
+        """processes metadata annotations
+
+        Args:
+            dmge (DataModelGraphExplorer): data model graph explorer
+            metadata_syn (dict): metadata used for Synapse storage
+            hideBlanks (bool): if true, does not upload annotation keys with blank values.
+            csv_list_regex (str):
+            annos (dict):
+            annotation_keys (str): display_label/class_label
+
+        Returns:
+            dict: annotations as a dictionary
+        """
+        for anno_k, anno_v in metadata_syn.items():
+            # Remove keys with nan or empty string values from dict of annotations to be uploaded
+            # if present on current data annotation
+            if hide_blanks and (
+                (isinstance(anno_v, str) and anno_v.strip() == "")
+                or (isinstance(anno_v, float) and np.isnan(anno_v))
+            ):
+                annos.pop(anno_k) if anno_k in annos.keys() else annos
```

Comment:
I don't understand this line, could you include a comment?

Reply:
the comment was added in line 1543: https://github.com/Sage-Bionetworks/schematic/pull/1478/files#diff-7b756ab3c7de9e192e644befb74f09e376c916bb68bc3aefb66989510371eba3R1543

I also kept this part the same as the original function: https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/store/synapse.py#L1583